src/utils/communication_buffer.hpp

```diff
@@ -0,0 +1,418 @@
+//========================================================================================
+// (C) (or copyright) 2020-2022. Triad National Security, LLC. All rights reserved.
+//
+// This program was produced under U.S. Government contract 89233218CNA000001 for Los
+// Alamos National Laboratory (LANL), which is operated by Triad National Security, LLC
+// for the U.S. Department of Energy/National Nuclear Security Administration. All rights
+// in the program are reserved by Triad National Security, LLC, and the U.S. Department
+// of Energy/National Nuclear Security Administration. The Government is granted for
+// itself and others acting on its behalf a nonexclusive, paid-up, irrevocable worldwide
+// license in this material to reproduce, prepare derivative works, distribute copies to
+// the public, perform publicly and display publicly, and to permit others to do so.
+//========================================================================================
+
+#ifndef UTILS_COMMUNICATION_BUFFER_HPP_
+#define UTILS_COMMUNICATION_BUFFER_HPP_
+
+#include <iostream>
+#include <limits>
+#include <memory>
+#include <type_traits>
+#include <unordered_map>
+#include <utility>
+
+#include "utils/mpi_types.hpp"
+
+#ifdef MPI_PARALLEL
+#include <mpi.h>
```

Comment:
```suggestion
```
Should probably be replaced by `#include <parthenon_mpi.hpp>` outside of the #ifdef here